interview-questions.md

```diff
@@ -10,58 +10,71 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is `this` in JavaScript?
 
   Your answer:
+"This" is a javascript keyword that refers an object. It can be used with .notation method to access a value that's within in object. The method, or the function within an object can refer to its values using the key and "this", like "this.(name of key)"
 
   Researched answer:
-
+The keyword "this" is a reference to an object but which object depends on how this is invoked. For example "this" by itself or in a function refers to the global object, while this with methods refer this to any object.
 
 
 2. What is React? Why would you use it?
 
   Your answer:
+  React is an open source javascript library used to build user interfaces. It is widely used in the industry and maintained by Meta (originally Facebook). React is very modular and based on components which are separate reusable pieces of code. It saves a lot of time for building the front-end is very good for making web applications with dynamic data.
 
   Researched answer:
-
+  React is a javascript based user interface development library and is widely used among web developers. It allows easy creation of dynamic applications with pre-built code (less coding). React's Virtual DOM (or document object model) is a highly performing and efficient way to render data by comparing its components previous states and updates items that the DOM that were changed, leaving the other components untouched.
 
 
 3. Which lifecycle method is required in a React class component?
 
   Your answer:
+  The constructor method is required when creating data that belongs to a class. It runs automatically when a class is instantiated. For example when you're setting up a component and instantiating the class, you would type "constructor(){} within the class's code block.
 
   Researched answer:
+During the "Mounting" component lifecycle, the constructor method is required and called before anything else in the component. It's called with "props", as arguments and should always call super(props) before anything else which will initiate the parent class's constructor method and allows the component to inherit methods from the parent class.
```

Comment:
That is the proper way to make a space to hold data (state object). However, not all class components store data. Some components only display information and will not require any storage. Yet all components will need a render method.